src/main/java/hudson/plugins/git/extensions/impl/CloneOption.java

```diff
@@ -148,7 +149,20 @@ public void decorateCloneCommand(GitSCM scm, Run<?, ?> build, GitClient git, Tas
             // in a single job definition.
             RemoteConfig rc = scm.getRepositories().get(0);
             List<RefSpec> refspecs = rc.getFetchRefSpecs();
-            cmd.refspecs(refspecs);
+            List<RefSpec> expandedrefSpecs = new ArrayList<>();
+            Boolean check = false;
+            for (RefSpec ref:refspecs) {
+                if(ref.toString().contains("$")){
+                    EnvVars env = build.getEnvironment(listener);
```

Comment:
Shouldn't this be done outside the loop?

Reply:
I have changed the implementation and shifted this outside.